.NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseDateTimeAltExtractor.cs

```diff
@@ -367,42 +352,118 @@ private bool IsConnectorOrWhiteSpace(int start, int end, string text)
+                    // RelativePrefix as Context: such as "next Tuesday or Wednesday", "next Monday or Tuesday or previous Wednesday"
+                    int i = 0;
+                    ExtractResult contextEr = ExtractRelativePrefixContext(extractResults[0].Text);
+
+                    while (i < extractResults.Count)
                     {
-                        var match = regex.Match(extractResults[0].Text);
-                        if (match.Success)
+                        if (contextEr == null)
+                        {
+                            break;
+                        }
+                        else
+                        {
+                            extractResults[i].Data = GenerateAltData(extractResults[i].Type, parentText, contextEr: null);
+                            extractResults[i].Type = ExtractorName;
+
+                            isValidContext = true;
+                        }
+
+                        int j = i + 1;
+
+                        while (j < extractResults.Count)
                         {
-                            var contextErs = new ExtractResult
+                            var relativePrefixContext = ExtractRelativePrefixContext(extractResults[j].Text);
+
+                            // No relativePrefix extracted, the context would follow the previous relativePrefix
+                            // Such as "Wednesday" in "next Tuesday or Wednesday"
+                            if (relativePrefixContext == null)
+                            {
+                                extractResults[j].Data = GenerateAltData(extractResults[j].Type, parentText, contextEr);
+                                extractResults[j].Type = ExtractorName;
+                                j++;
+                            }
+                            else
                             {
-                                Text = match.Value,
-                                Start = match.Index,
-                                Length = match.Length,
-                                Type = Constants.ContextType_RelativePrefix
-                            };
-
-                            data.Add(Constants.Context, contextErs);
-                            data.Add(Constants.SubType, Constants.SYS_DATETIME_DATE);
+                                // Current extraction has relativePrefix, the context would not follow the previous relativePrefix
+                                // Such as "Wednesday" in "next Monday or Tuesday or previous Wednesday"
+                                contextEr = relativePrefixContext;
+                                break;
+                            }
                         }
+
+                        i = j;
                     }
                 }
             }
-            
+
+            return isValidContext;
+        }
+
+        private Dictionary<string, object> GenerateAltData(string subType, string parentText, ExtractResult contextEr = null)
```

Comment:
I don't get what this means.

Reply:
Changed name of this method to "CreateMetadata"